### DISCUSSION TITLE
Can I remap every key? (migrating from neovim)

### DISCUSSION THREAD

**Discussion by julien-h2** [helix-editor/helix#11743] [Q&A] [answered]
Hello, I'm looking into Helix and might consider migrating from neovim. As I'm thinking it might be a faster software free from the past.

**Question 1:** As I use a non-qwerty keyboard I have remapped every key in my neovim configuration to something custom. Is that something that Helix allows reliably?

**Question 2:** is there some scripting language where I can define custom actions and textobjects?

**Questin 3:** Is there any equivalent to flash.nvim? To jump/select treesitter nodes with labels

**Comment by kirawi** [comment, marked as answer]
1. To an extent. There are some special-cased commands that need to handle their own key input (such as `mi` or `ma`) and thus cannot be rebound [1].
2. [Not yet.](https://github.com/helix-editor/helix/pull/8675)
3. Only Amp-like jump-mode is implemented via `gw` which only jumps to words, not tree-sitter nodes to keep the feature simple.


[1] https://github.com/helix-editor/helix/issues/9814#issuecomment-1982097452

**Reply by julien-h2** [reply]
Thank you, I'll wait a bit more before jumping in but this project is very exciting
